applications/ContactStructuralMechanicsApplication/python_scripts/contact_structural_mechanics_analysis.py

```diff
@@ -83,32 +69,36 @@ def _CreateSolver(self, external_model_part=None):
             ## Read the model - note that SetBufferSize is done here
             self.solver.ReadModelPart() # TODO move to global instance
 
-    def _ExecuteInitialize(self):
-        """ Initializing the Analysis """
-
-        super()._ExecuteInitialize()
-
-        ## Processes construction
-        import process_factory
-        if (self.ProjectParameters.Has("contact_process_list") is True):
-            self.list_of_processes += process_factory.KratosProcessFactory(self.structure_model).ConstructListOfProcesses(self.ProjectParameters["contact_process_list"])
-
-            #if self.is_printing_rank and self.echo_level > 1: # FIXME
-                #KM.Logger.PrintInfo("Process " + str(len(self.list_of_processes)), self.list_of_processes[-1])
-
-            self.list_of_processes[-1].ExecuteInitialize()
-
-        ## Add the processes to the solver
-        self.solver.AddProcessesList(self.list_of_processes)
-        if (self.output_post is True):
-            self.solver.AddPostProcess(self.gid_output)
-
-        # Initialize the solver (again)
-        self.solver.Initialize()
-
-        # Setting the echo level
-        echo_level = self.ProjectParameters["problem_data"]["echo_level"].GetInt()
-        self.solver.SetEchoLevel(echo_level)
+    def _SetUpListOfProcesses(self):
+        """ Set up the list of processes """
+
+        from process_factory import KratosProcessFactory
```

Comment:
This is a lot of duplication, why don't you call the baseclass and the add the contact process, same as what you are doing now?

Reply:
Because just after creating the list some others operations are used